### PR TITLE
Fix checkRestrictions

### PR DESCRIPTION
**Pull request by bitPogo**
## Description
This addresses the issue, which had been experienced with the [Patient testset](https://github.com/gesundheitscloud/s4h-fhir-profiles/blob/94c4afe43e60bf5c8c4faa5c67276d9ce533789f/examples/s4h-patient-example.patient.json#L1).

## Motivation and Context
The current implementation for `checkDataRestrictions` breaks if a Resource contains Attachments (or at least seen by the FhirAttachmentHelper as such), which are not extractible out of the Resource.

## How is it being implemented?
The Attachements are not longer referenced as List, there are now nullable. Additionally after the extraction of the Attachments is now a null check and if the Attachments are null `checkDataRestrictions` simply returns.
Also to avoid any confusion the style of `_uploadData` and `checkDataRestrictions` had been aligned.

## How Has This Been Tested?
A new Unittest had been added, which makes use of the [Patient testset](https://github.com/gesundheitscloud/s4h-fhir-profiles/blob/94c4afe43e60bf5c8c4faa5c67276d9ce533789f/examples/s4h-patient-example.patient.json#L1), which breaks the current SDK.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sdk-core/src/test/java/care/data4life/sdk/RecordServiceTest.kt [hunk 1]
```diff
@@ -823,6 +825,79 @@ class RecordServiceTest : RecordServiceTestBase() {
         inOrder.verifyNoMoreInteractions()
     }
 
+    @Test
+    @Throws(DataRestrictionException.UnsupportedFileType::class, DataRestrictionException.MaxDataSizeViolation::class)
+    fun `Given, checkDataRestrictions is called, with a Resource, which has non extractable Attachments, it returns without a failure`() {
+        // Given
+        val resourceStr = "{\n" +
+                "    \"resourceType\": \"Patient\",\n" +
```
Comment (wmontwe) [inline sdk-core/src/test/java/care/data4life/sdk/RecordServiceTest.kt:833]: I think the example payload should be shrinked to a minimal version, otherwise there is a lot of information that is not related to this check

**Review by wmontwe — CHANGES_REQUESTED** [review]
Good fix 👍 Just the example payload is a little to big.

File: sdk-core/src/test/java/care/data4life/sdk/RecordServiceTest.kt (hunk 1 above)
Comment (wmontwe) [reply sdk-core/src/test/java/care/data4life/sdk/RecordServiceTest.kt:833]: or the resource is moved to the resource folder

Comment (bitPogo) [reply sdk-core/src/test/java/care/data4life/sdk/RecordServiceTest.kt]: I pulled it out in the resource folder...However, since we will refactor the test soonish, I will invest a bit more time to think of a more generic example then, to decouple the test.

File: sdk-core/src/test/java/care/data4life/sdk/RecordServiceTest.kt [hunk 2]
```diff
@@ -823,6 +833,22 @@ class RecordServiceTest : RecordServiceTestBase() {
         inOrder.verifyNoMoreInteractions()
     }
 
+    @Test
+    @Throws(DataRestrictionException.UnsupportedFileType::class, DataRestrictionException.MaxDataSizeViolation::class)
+    fun `Given, checkDataRestrictions is called, with a Resource, which has non extractable Attachments, it returns without a failure`() {
+        // Given
+        val resourceStr = getResource("/fhir4/s4h-patient-example.patient.json")
```
Comment (wmontwe) [inline sdk-core/src/test/java/care/data4life/sdk/RecordServiceTest.kt:840]: 👍

**Review by wmontwe — APPROVED** [review]
👍
